File: Projects/CPP-VNEXT/src/Plants/Guarana.cpp
#include "Guarana.h"
#include "../World.h"

Guarana::Guarana(Position&& position) : Plant(0, "\033[38;5;160m&\033[m", std::move(position)) {}

const std::string Guarana::Type = "Guarana";

std::string Guarana::GetType() const {
    return Guarana::Type;
}

Organism* Guarana::GetNewOfType(Position&& position) {
    return new Guarana(std::move(position));
}

void Guarana::HandleCollision(CollisionContext& context) {
    context.GetAttacker()->IncreaseStrength(3);
    context.KillHost();
}

File: Projects/Java/src/main/java/com/nieradko/worldsim/core/plants/Guarana.java
package com.nieradko.worldsim.core.plants;

import com.nieradko.worldsim.core.ICollisionContext;
import com.nieradko.worldsim.core.IWorldContext;
import com.nieradko.worldsim.core.Position;
import javafx.scene.paint.Color;

public class Guarana extends Plant {
    public Guarana(Position position) {
        super(0, Color.ORANGERED, position);
    }

    @Override
    protected void handleCollision(ICollisionContext collisionContext, IWorldContext worldContext) {
        worldContext.log(String.format("%s has been consumed which increased %s's strength", getName(), collisionContext.getAttacker().getName()));
        collisionContext.getAttacker().increaseStrength(3);
        collisionContext.defenderHasDied();
    }
}
